### ISSUE TITLE
Cannot call Task.finalizedBy() after task has started execution

### ISSUE THREAD

**Issue by stephanedupont** [bugsnag/bugsnag-android-gradle-plugin#287]
I am in the process of migrating from Crashlytics to Bugsnap for some Android apps.

Adding the bugsnap android gradle plugin in the project result in the following error when I run a debug build from Android Studio:

`What went wrong:
Could not create task ':app:processBugsnagGoogle-debugManifest'.
Cannot call Task.finalizedBy(Object...) on task ':completeRhythmTrainer:processGoogleDebugManifest' after task has started execution.`

- If I remove the bugsnap plugin, the error message disappears.
- If I build a release flavor and not a debug flavor, the error message disappears

**Comment by xander-jones** [comment]
Hey @stephanedupont, thank you for reporting this bug. It's an issue that we're aware of, and we're looking to fix. We'll keep this thread updated with changes.

**Comment by pbajurko** [comment]
For me it happens only when building from AS but not form gradle command. Still it's integration blocking issue. @fractalwrench when can we expect next plugin release with fix? Do we have any snapshot releases?

**Comment by fractalwrench** [comment]
We believe this issue occurs for users who run their app in Debug mode in Android Studio and are using Android Gradle Plugin v4.0.

We are investigating this as a matter of priority, but we have encountered some issues in finding a solution that works for everyone. We'll post an update once we have a full solution but in the meantime, as you say, you can use the command line or you can upgrade to 4.1.0 of the Android Gradle Plugin (although this is currently in RC). Another workaround is to disable the plugin for the debug buildType (along with any other build variants that don't require deobfuscated stacktraces in Bugsnag):

```groovy
bugsnag {
    variantFilter { variant ->
        // disables plugin for all variants of the 'debug' buildType
        def name = variant.name.toLowerCase()
        if (name.contains("debug")) {
            enabled = false
        }
    }
}
```

**Comment by pbajurko** [comment]
Well problem is that this plugin also delivers Bugsnag native libs required for compilation(It should be done by separate plugin or at least it should not depend on enabled flag). And dropping library modules support in v5 didn't make it easier :) So I would like to try your latest fix since it should be enough for us. 

Btw I also tried AS 4.0 with Bugsnag example app(AGP 3.6) and plugin 5.0.1 and issue is also there. So it seems that issue somehow related to AS only?

**Comment by phillipsam** [comment]
A fix for this has now been released in [v5.0.2](https://github.com/bugsnag/bugsnag-android-gradle-plugin/releases/tag/v5.0.2).
